### ISSUE TITLE
Support Polish

### ISSUE THREAD

**Issue by saattrupdan** [EuroEval/EuroEval#704]
Potential Polish datasets to fit into the existing tasks:

- Reading comprehension: [PoQuAD](https://huggingface.co/datasets/clarin-pl/poquad) or [PoQuAD v2](https://huggingface.co/datasets/mzasada/poquad_v2), maybe [PolQA](https://huggingface.co/datasets/ipipan/polqa)
- Sentiment classification: [PolEmo2.0](https://huggingface.co/datasets/clarin-pl/polemo2) (both the 'in' and 'out' subset) and/or the [CLARIN-Emo dataset](https://huggingface.co/datasets/clarin-knext/CLARIN-Emo).
- Linguistic acceptability: ScaLA algorithm with [Polish universal dependencies](https://github.com/UniversalDependencies/UD_Polish-PDB)
- Named entity recognition: [KPWR-NER](https://huggingface.co/datasets/clarin-pl/kpwr-ner)
- Summarization: [PolSum](https://huggingface.co/datasets/maciej-ogrodniczuk/polsum), [PSC](https://huggingface.co/datasets/community-datasets/psc) and [Polish News](https://huggingface.co/datasets/WiktorS/polish-news)
- Knowledge: [LLMzSzL](https://huggingface.co/datasets/amu-cai/llmzszl-dataset)
- Reasoning: ?
- Potentially NLI datasets (would need to be set up): [SICK-pl](https://huggingface.co/datasets/sdadas/sick_pl) and [CDSCorpus](https://zil.ipipan.waw.pl/Scwad/CDSCorpus). Could potentially replace the common-sense reasoning.

**Comment by saattrupdan** [comment]
It seems like there is no translation of [HellaSwag](https://huggingface.co/datasets/alexandrainst/m_hellaswag/viewer/en) to Polish. With that, it seems that we have enough available datasets to add it to the benchmark.

**Comment by djstrong** [comment]
> * Reading comprehension: [PoQuAD](https://huggingface.co/datasets/clarin-pl/poquad) or [PoQuAD v2](https://huggingface.co/datasets/mzasada/poquad_v2), maybe [PolQA](https://huggingface.co/datasets/ipipan/polqa)

These are good datasets (present on https://huggingface.co/spaces/speakleash/open_pl_llm_leaderboard). One problem is that the answer is very short: 1-3 words and many LLMs have problems with answering shorter than one sentence :)

> * Sentiment classification: [PolEmo2.0](https://huggingface.co/datasets/clarin-pl/polemo2)

There is two parts of this dataset: `in` and `out`. Training split is only available for `in` domain. Both are on https://huggingface.co/spaces/speakleash/open_pl_llm_leaderboard

> * Named entity recognition: [NJKP NER](https://huggingface.co/datasets/nkjp/nkjp-ner)

I think it shouldn't be used because classes are very complicated and need detailed description in a prompt, maybe with examples. The best model on https://huggingface.co/spaces/speakleash/open_pl_llm_leaderboard achieve 65% but small models with a little finetuning about 80-90%.

> * Summarization: [PolSum](https://huggingface.co/datasets/maciej-ogrodniczuk/polsum), [PSC](https://huggingface.co/datasets/community-datasets/psc) and [Polish News](https://huggingface.co/datasets/WiktorS/polish-news)

PSC is on https://huggingface.co/spaces/speakleash/open_pl_llm_leaderboard

> * Knowledge: [GlobalMMLU-pl](https://huggingface.co/datasets/CohereForAI/Global-MMLU)

GlobalMMLU-pl is machine translated. Maybe you can use Poquad or PolQA without context?

> * Reasoning: ?

What is English example of this task?


https://zil.ipipan.waw.pl/Scwad/CDSCorpus , https://huggingface.co/datasets/sdadas/sick_pl - entailment datasets

**Comment by saattrupdan** [comment]
> > * Reading comprehension: [PoQuAD](https://huggingface.co/datasets/clarin-pl/poquad) or [PoQuAD v2](https://huggingface.co/datasets/mzasada/poquad_v2), maybe [PolQA](https://huggingface.co/datasets/ipipan/polqa)
> 
> These are good datasets (present on https://huggingface.co/spaces/speakleash/open_pl_llm_leaderboard). One problem is that the answer is very short: 1-3 words and many LLMs have problems with answering shorter than one sentence :)

Good to hear that they're of good quality! And regarding the short answers, we found that including "Answer in less than 3 words" worked surprisingly well for many model sizes, but we haven't conducted a systematic study of this yet.

> > * Sentiment classification: [PolEmo2.0](https://huggingface.co/datasets/clarin-pl/polemo2)
> 
> There is two parts of this dataset: `in` and `out`. Training split is only available for `in` domain. Both are on https://huggingface.co/spaces/speakleash/open_pl_llm_leaderboard

That's great to hear, and I see that the default is the in-domain evaluation. Would you say that it could make sense to mix the two datasets? That would also increase the size of the test set, which I see is 723 for the in-domain subset and 494 for the out-of-domain subset.

> > * Named entity recognition: [NJKP NER](https://huggingface.co/datasets/nkjp/nkjp-ner)
> 
> I think it shouldn't be used because classes are very complicated and need detailed description in a prompt, maybe with examples. The best model on https://huggingface.co/spaces/speakleash/open_pl_llm_leaderboard achieve 65% but small models with a little finetuning about 80-90%.

Upon closer inspection, I see that the task doesn't have any token labels either, making it unsuitable for token classification (must've just been misclassified on the Hugging Face Hub).

Are there any token-level NER tasks in Polish, that you're aware of?

> > * Knowledge: [GlobalMMLU-pl](https://huggingface.co/datasets/CohereForAI/Global-MMLU)
> 
> GlobalMMLU-pl is machine translated. Maybe you can use Poquad or PolQA without context?

Potentially, but the problem is that the knowledge task is set up as a multiple choice task currently. That being said, we could potentially add a new abstractive question answering task and have this as an example. We would probably need to filter the data as well, to make sure that we're testing knowledge (ideally also noting roughly what kind of knowledge).

> > * Reasoning: ?
> 
> What is English example of this task?

In English the HellaSwag dataset is used, see more [here](https://euroeval.com/datasets/english/#hellaswag).

> https://zil.ipipan.waw.pl/Scwad/CDSCorpus , https://huggingface.co/datasets/sdadas/sick_pl - entailment datasets

Thanks! These kind of NLI datasets could also potentially be included. We would need create a new task by tweaking an existing one.

**Comment by djstrong** [comment]
> > > * Sentiment classification: [PolEmo2.0](https://huggingface.co/datasets/clarin-pl/polemo2)
> > 
> > 
> > There is two parts of this dataset: `in` and `out`. Training split is only available for `in` domain. Both are on https://huggingface.co/spaces/speakleash/open_pl_llm_leaderboard
> 
> That's great to hear, and I see that the default is the in-domain evaluation. Would you say that it could make sense to mix the two datasets? That would also increase the size of the test set, which I see is 723 for the in-domain subset and 494 for the out-of-domain subset.
> 

I wouldn't say in-domain evaluation is the default. I think you can mix these two. Here is also sentiment classification data: https://poleval.pl/tasks/task2

> > > * Named entity recognition: [NJKP NER](https://huggingface.co/datasets/nkjp/nkjp-ner)
> > 
> > 
> > I think it shouldn't be used because classes are very complicated and need detailed description in a prompt, maybe with examples. The best model on https://huggingface.co/spaces/speakleash/open_pl_llm_leaderboard achieve 65% but small models with a little finetuning about 80-90%.
> 
> Upon closer inspection, I see that the task doesn't have any token labels either, making it unsuitable for token classification (must've just been misclassified on the Hugging Face Hub).
> 
> Are there any token-level NER tasks in Polish, that you're aware of?
> 

This dataset was converted from token level classification to text classification. It is based on annotated corpus NKJP. It is possible to extract token-level annotation from it but it needs more research to be suitable  from LLM evaluation.

Here is another dataset but with 80 classes: https://huggingface.co/datasets/clarin-pl/kpwr-ner

> > > * Knowledge: [GlobalMMLU-pl](https://huggingface.co/datasets/CohereForAI/Global-MMLU)
> > 
> > 
> > GlobalMMLU-pl is machine translated. Maybe you can use Poquad or PolQA without context?
> 
> Potentially, but the problem is that the knowledge task is set up as a multiple choice task currently. That being said, we could potentially add a new abstractive question answering task and have this as an example. We would probably need to filter the data as well, to make sure that we're testing knowledge (ideally also noting roughly what kind of knowledge).
> 

https://huggingface.co/spaces/amu-cai/LLMZSZL_Leaderboard is constructed from school tests

**Comment by saattrupdan** [comment]
> I wouldn't say in-domain evaluation is the default. I think you can mix these two. Here is also sentiment classification data: https://poleval.pl/tasks/task2

Oh, I mean that the actual default when loading that dataset _is_ the in-domain data - it's in [this line](https://huggingface.co/datasets/clarin-pl/polemo2/blob/ecd39c467892f767733c9368ccd2148c2094f472/polemo2.py#L73) of the dataset generation script.

> Here is another dataset but with 80 classes: https://huggingface.co/datasets/clarin-pl/kpwr-ner

That's great, thanks!

> https://huggingface.co/spaces/amu-cai/LLMZSZL_Leaderboard is constructed from school tests

This is an amazing dataset, that fits perfectly within the knowledge dataset, thanks for pointing us to this! The concrete dataset used in this leaderboard seems to be [this one](https://huggingface.co/datasets/amu-cai/llmzszl-dataset), which can be directly used as a Polish knowledge dataset 😊

**Comment by saattrupdan** [comment]
Relevant knowledge dataset:
https://arxiv.org/abs/2503.00995
